### PR TITLE
evaluate: accept options argument

### PR DESCRIPTION
**Pull request by michaelfig**
Notably, accept `options.transforms` from [realms-shim](https://github.com/Agoric/realms-shim).

This version changes the package name to `@agoric/insecure-evaluate`, and introduces exported `evaluteExpr`, `evaluateProgram`, and `makeEvaluators`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,9 @@
 # Evaluate
 
-This repository is NOT A SECURE IMPLEMENTATION!
+INSECURE three-argument evaluate function.
 
-You should be using [SES](https://github.com/Agoric/SES) or
-[realms-shim](https://github.com/Agoric/realms-shim) to call `evaluate`
-with secure control of global variables.
+You should be using [Secure ECMAScript](https://github.com/Agoric/SES) or Realms
+to call `evaluate` with security in mind.
```
Comment (warner) [inline README.md:8]: maybe add a note to explain what we mean by insecure here. "The evaluated code will have full access to the globals and something something scope of the caller, which is far more authority than you really want to give that code".

File: src/index.js [hunk 2]
```diff
@@ -1,11 +1,46 @@
...
+// The evaluate maker, which curries the makerOptions.
+export const makeEvaluators = (makerOptions = {}) => {
+  const makeEvaluator = isExpr => (source, endowments = {}, options = {}) => {
+    const fullTransforms = [
+      ...(options.transforms || []),
+      ...(makerOptions.transforms || []),
+    ];
+    const fullEndowments = {
+      ...(makerOptions.endowments || {}),
+      ...endowments,
```
Comment (warner) [inline src/index.js:10]: just doublechecking.. the local `endowments` will override the ones from `makerOptions`, right?

**Review by warner — APPROVED** [review]
Looks good. I don't know how we'll publish/use this module, but I'll bet you've got a plan, and I suspect it'll lead to a cleaner solution than our current `@agoric/evaluate` tarball, so I'm looking forward to seeing it :).

File: src/index.js (hunk 2 above)
Comment (michaelfig) [reply src/index.js:10]: Yes, right.

File: README.md (hunk 1 above)
Comment (michaelfig) [reply README.md:8]: Thanks, this encouraged me to prevent one form of leakage.

**Pull request merged**